source/libs/on-pr-file-load.tsx

```diff
@@ -0,0 +1,28 @@
+import select from 'select-dom';
+import debounce from 'debounce-fn';
+import domLoaded from 'dom-loaded';
+
+const handlers = new Set<VoidFunction>();
+
+const fragmentsLoaded = debounce(() => {
+	// Safely run all callbacks
+	handlers.forEach(async cb => cb());
+}, {wait: 200});
+
+function setup(): void {
+	for (const fragment of select.all('include-fragment.diff-progressive-loader')) {
+		// Remove any existing event listeners
+		fragment.removeEventListener('load', fragmentsLoaded);
```

Comment:
> `addEventListener` doesn’t add the same handler two, so it doesn’t need to be removed.

Done

------------------

> Also this handler shouldn’t be too complicated, the whole file should just be a default export function with the loop that adds the callback as a listener.

Done

------------------

> Also it probably shouldn’t be part of `features`’ loaders because it’s never used on its own. You can leave plain call in your feature’s `init` function rather than using that new `load`

That piece of code was extracted out as you said, from `raw-file-link`, so that that part of code can be reused (with the intention of updating `raw-file-link` later).

This feature need to be triggered on both new comments and diff loads, but raw-file-link only depends on diff loads, hence the need for this new load mechanism.